assets/js/amp-customize-controls.js

```diff
@@ -23,10 +31,13 @@
 	 * @return {string} non-AMPified URL.
 	 */
 	function unampifyUrl( url ) {
-		var urlParser = document.createElement( 'a' );
+		var urlParser = document.createElement( 'a' ),
+			regexParam = new RegExp( "(^|\\?|&)" + ampVars.query + "=1" ),
+			regexEndpoint = new RegExp( "\\/" + ampVars.query + "\\/?$" );
+
 		urlParser.href = url;
-		urlParser.pathname = urlParser.pathname.replace( /\/amp\/?$/, '' );
-		urlParser.search = urlParser.search.replace( /(^|\?|&)amp=1/, '' );
+		urlParser.pathname = urlParser.pathname.replace( regexEndpoint, '' );
+		urlParser.search = urlParser.search.replace( regexParam, '' );
```

Comment:
Per above, instead of `regexParam` I suggest doing something like:

```
if ( urlParser.search.length > 1 ) {
    var params = wp.customize.utils.parseQueryString( urlParser.search.substr( 1 );
    delete params[ ampVars.query ];
    urlParser.search = $.param( params );
}
```

This is similar to what is done in core: https://github.com/WordPress/wordpress-develop/blob/72b74509f3bfb71c884bf8cd9f580c026079498a/src/wp-admin/js/customize-controls.js#L6547-L6559